org.eclipse.winery.frontends/app/tosca-management/src/app/instance/serviceTemplates/researchObject/researchObject.service.ts

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2021-2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpClient, HttpHeaders, HttpResponse } from '@angular/common/http';
+import { ROMetadataApiData, ROPublicationApiData } from '../../../model/researchObjectApiData';
+import { FileOrFolderElement } from '../../../model/fileOrFolderElement';
+import { MoveElementApiData } from '../../../model/moveElementApiData';
+
+
+@Injectable()
+export class ResearchObjectService {
+
+    researchObjectMetadata: ROMetadataApiData;
+    researchObjectPublication: ROPublicationApiData;
+    pathToElementsMap: Map<string, FileOrFolderElement[]>;
+    url: string;
+
+    constructor(private http: HttpClient) {
+    }
+
+    getResearchObjectMetadata(): Observable<ROMetadataApiData> {
+        const o = this.http.get<ROMetadataApiData>(this.url + 'metadata');
+        o.subscribe(data => this.researchObjectMetadata = data);
```

Comment:
We took our design from other, already existing components, e.g., the [SelfService](https://github.com/eclipse/winery/blob/main/org.eclipse.winery.frontends/app/tosca-management/src/app/instance/serviceTemplates/selfServicePortal/selfServicePortal.service.ts) component, where it is done in a similar way. I think it makes sense this way because of the child components that fetch the data from here.